pylink/jlink.py

```diff
@@ -3067,7 +3081,11 @@ def register_read_multiple(self, register_indices):
         Raises:
           JLinkException: if a given register is invalid or an error occurs.
         """
+        # TODO: rename 'register_indices' to 'registers'
         num_regs = len(register_indices)
+        for idx, indice in enumerate(register_indices):
+            if isinstance(indice, six.string_types):
+                register_indices[idx] = self._get_register_index_from_name(indice)
```

Comment:
This is mutative (has a side effect).  I'd either create a new list and copy it over to it, or before entering this for-loop, do a copy such as:
```
register_indices = register_indices[:]
```